app/controllers/sessions_controller.rb

```diff
@@ -4,11 +4,23 @@ class SessionsController < ApplicationController
   def create
     auth_hash = request.env['omniauth.auth']
 
-    if auth_hash["uid"]
+    if auth_hash && auth_hash["uid"]
       @user = User.find_or_create_from_omniauth(auth_hash)
```

Comment:
Ok - perfect. Thank you for the clarification. I wasn't sure if you were still planning to do the CSV portion as well.